torch/csrc/cuda/lazy_init.h

```diff
@@ -0,0 +1,21 @@
+#include <Python.h>
+#include <mutex>
+
+#include "torch/csrc/Exceptions.h"
+#include "torch/csrc/utils/object_ptr.h"
+
+namespace torch {
+namespace cuda {
+
+static void lazy_init() {
+  static std::once_flag once;
```

Comment:
This will create a `once_flag` per .cpp file that includes this header. You might be able to mark it `extern` and define it in a `.cpp` file.

I think the function should be marked `inline` since it's defined in the header.